### ISSUE TITLE
Suggestion about OpenDAL's Reader API

### ISSUE THREAD

**Issue by Xuanwo** [xrelkd/caracal#25]
https://github.com/xrelkd/caracal/blob/8843f711209272607750686f62d80f5d4403d55f/crates/engine/src/fetcher/minio.rs#L73

Hi, thanks for using OpenDAL! I start this issue to want you know that `opendal` has native range support so that you can:

```rust
self.operator
    .reader_with(self.filename.as_str())
    .range(start..end)
    .await
    .context(error::CreateReaderSnafu)?;
```

Hoping this will be helpful!

**Comment by xrelkd** [comment]
Wow!!
@Xuanwo Thanks for the suggestion of using `opendal::Operator::reader_with`, it's really helpful.

Good to see a `Rust` library/framework for accessing different storage services, database services, etc..
`OpenDAL` is a nice and amazing library, it does save time and hassle.

This is resolved by #26.

**Comment by Xuanwo** [comment]
Glad to help!
